### ISSUE TITLE
Exclude specific gateway in heterogeneous cluster?

### ISSUE THREAD

**Issue by wangjia184** [dotnet/orleans#7643]
I have a heterogeneous cluster with two groups of silos,  Group A and Group B

There is a serializable type served by silos added in Group B only. 

As the [document ](https://dotnet.github.io/orleans/docs/host/heterogeneous_silos.html)

> All silos should reference interfaces of all grain types of the cluster, but grain classes should only be referenced by the silos that will host them.

Certainly I can make it to work by deploying all silos with this new serializable type.

But I don't want to deploy Group A because they are just dependency of Group B.  
If my client connect to silo's of Group A, which behaves as a gateway there, it would raise an exception because this new serializable type is unknown to it.
If my client only connects to silos of Group B, even this new type does not deploy in Group A,  it works well

So, is there a way for client to select gateway?
Or it is possible for silos of group A to disable their gateway?

```
siloBuilder.Configure<EndpointOptions>(options =>
                    {
                         options.GatewayPort = 0; // what if I change the advertised gateway port to invalid port so that client cannot connect?
                    });
```

**Comment by benjaminpetit** [comment]
As you mention, setting `GatewayPort` to zero will disable the silo gateway, so if you don't want to redeploy Group A, that's a solution

**Comment by wangjia184** [comment]
Hi @benjaminpetit 

From my test, this does not work

```
.Configure<EndpointOptions>(options =>
                    {
                        // announce port to use for the gateway
                        options.GatewayPort = 0; // dont want others to see me gateway 
                        // listenning port
                        options.GatewayListeningEndpoint = new IPEndPoint(IPAddress.Any, gatewayPort);

                        // IP Address to advertise in the cluster
                        options.AdvertisedIPAddress = announcedIP;
                    })
```

**Comment by wangjia184** [comment]
gatewayPort is a valid number,  while `GatewayPort` is zero.  you mean I should listen on port zero?  let me try again

**Comment by wangjia184** [comment]
yep,  announce `GatewayPort`  zero is not enough,  GatewayListeningEndpoint listens on port zero does not raise an error.
